### PR TITLE
fix: Missing Proguard rules for R8 in full mode

### PR DESCRIPTION
**Pull request by azlekov**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Android/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Android/issues?q=is%3Aissue).

### Issue Description

Closes: #1194

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by mtrezza** [comment]
Is this ready for merge?

**Review by mtrezza — APPROVED** [review]
Looks good

**Comment by azlekov** [comment]
> Looks good
> 
> 
> 
> Is this ready for merge? I've changed this to a `feat` PR, since it seems to add support for Gradle 8 to the Parse SDK.

That's not quite correct. But fixes fine release issues with ProGuard and R8

**Comment by mtrezza** [comment]
I assumed so because the issues don't seem to have these occurred so far. Is this an adaptation for Gradle 8 or is this also a bug for Gradle <8?

**Comment by mtrezza** [comment]
Maybe we should start adding a gradle matrix to the CI where we stest with different gradle versions - would that make sense?

**Comment by azlekov** [comment]
Hey @mtrezza the issues which the developers are reporting are runtime. Because of wrong ProGuard rules after magnification is done Parse classes are erased and when you run the app it crashes. I need to think about if there can be some tests covering the cases by building and running simple app using parse via some instrumentation test and check whatever it crash or not.

**Comment by mtrezza** [comment]
Ok, so the minification is the core issue here. Other than that, wouod it still male sense to run the CI with multiple Gradle or Android versions? If so, I'll open a separate issue for that.

**Comment by mtrezza** [comment]
@azlekov This PR got somewhat stale, can we merge this or is there anything that is worth discussing beforehand?

File: parse/release-proguard.pro [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+   public <init>(...);
+}
 
 # Required for Parse
 -keepattributes *Annotation*
 -keepattributes Signature
+
 # https://github.com/square/okio#proguard
 -dontwarn okio.**
+-dontwarn org.codehaus.mojo.animal_sniffer.*
```
Comment (mtrezza) [inline parse/release-proguard.pro:13]: I assume this was added by mistake, or does the SDK have auch a dependency?

Comment (azlekov) [reply parse/release-proguard.pro]: Done, thanks!

**Comment by azlekov** [comment]
@mtrezza I change the title to be more precise, it's not a feat, but a fix for proguard R8 in full mode as mentioned in the #1194

**Comment by mtrezza** [comment]
Great! I've just slightly edited the title as it should describe the actual issue. Otherwise I think this is ready for merge?

**Review by mtrezza — APPROVED** [review]

**Comment by azlekov** [comment]
> Great! I've just slightly edited the title as it should describe the actual issue. Otherwise I think this is ready for merge?

Yes, go go go! 🚗

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.2.1](https://github.com/parse-community/Parse-SDK-Android/releases/tag/4.2.1)

**Pull request merged**